wpe/src/main/glue/browser/renderer_asurfacetransaction.cpp

```diff
@@ -29,37 +34,46 @@ RendererASurfaceTransaction::~RendererASurfaceTransaction()
     if (m_surface.window)
         ANativeWindow_release(m_surface.window);
 
-    m_lockedBuffer = nullptr;
+    // Release the stored exported buffer, if any, and if different from the locked buffer.
+    // If the same, the buffer will be released via the locked buffer.
+    if (m_state.exportedBuffer && m_state.exportedBuffer != m_state.lockedBuffer)
+        releaseBuffer(m_page.exportable(), *m_state.exportedBuffer);
+    // If locked buffer still exists, release it.
+    if (m_state.lockedBuffer)
+        releaseBuffer(m_page.exportable(), *m_state.lockedBuffer);
+    m_state = { };
 }
 
 void RendererASurfaceTransaction::surfaceCreated(ANativeWindow* window)
 {
+    // This is now the surface we work with. We also spawn the corresponding ASurfaceControl.
     m_surface.window = window;
     m_surface.control = ASurfaceControl_createFromWindow(m_surface.window, "RendererASurfaceTransaction");
 }
 
 void RendererASurfaceTransaction::surfaceChanged(int format, unsigned width, unsigned height)
 {
+    // Update the size.
+    // FIXME: currently neither the format nor the stored size are used anywhere.
```

Comment:
Could you file an issue and reference it in the comment, please?

Reply:
Done.